pkg/cmd/attestation/verify/verify_integration_test.go

```diff
@@ -138,6 +156,16 @@ func TestVerifyIntegrationReusableWorkflow(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("with repo and valid reusable signer workflow", func(t *testing.T) {
+		opts := baseOpts
+		opts.Owner = "malancas"
+		opts.Repo = "malancas/attest-demo"
+		opts.SignerWorkflow = "https://github.com/github/artifact-attestations-workflows/.github/workflows/attest.yml@09b495c3f12c7881b3cc17209a327792065c1a1d"
```

Comment:
Unrelated to the above, how (or should) pinning work?

I'm inclined to say we should leave ref off anything after the `@` for `--signer-worflow`. That content could be a git SHA (like it is in your example) or it could be a git ref (like `refs/heads/main`), depending on how the build was triggered. If it's a git ref (like `refs/heads/main`) there's no guarantee about what that ref points at now, which could be confusing for users.

If you want to ensure your build came from a specific SHA, you can do that by writing a policy, or by using `--cert-identity`.

Reply:
What if the user wants to provide a custom git ref (like a development branch) and a custom GitHub host? Would they use the `cert-identity` flag in that case?

Reply:
Fulcio captures the build signer digest OID `1.3.6.1.4.1.57264.1.10` which allows us to always reference that regardless how the build signer was referenced. So I think I'm with @steiza here to ignore the ref part, and only verify that the signer workflow originates from the specified repo.

Reply:
> Fulcio captures the build signer digest OID `1.3.6.1.4.1.57264.1.10` which allows us to always reference that regardless how the build signer was referenced. So I think I'm with @steiza here to ignore the ref part, and only verify that the signer workflow originates from the specified repo.

Do you think we should enforce that the user cannot specify a ref when calling this flag or accept it if the user includes one?

Reply:
> Do you think we should enforce that the user cannot specify a ref when calling this flag or accept it if the user includes one?

Ah, it could be okay to support it **if** the user supplies a SHA or ref, as long as we don't require it.

Reply:
Well, maybe not 😆 

I was thinking "oh, if someone passes `--signer-workflow` that exactly matches the SAN, shouldn't we accept that?"

Except, I don't think `--signer-workflow` works if you pass `https://` (and I don't think it necessarily should):

```
cli$ go run cmd/gh/main.go attestation verify -o actions --signer-workflow https://github.com/actions/attest-demo/.github/workflows/build-python.yml ~/demo/github_provenance_demo-0.0.12-py3-none-any.whl 
...
✗ Verification failed
```

```
$ go run cmd/gh/main.go attestation verify -o actions --signer-workflow github.com/actions/attest-demo/.github/workflows/build-python.yml ~/demo/github_provenance_demo-0.0.12-py3-none-any.whl 
...
✓ Verification succeeded!
```

So maybe we should leave this as-is.

Reply:
Yes, the `signer-workflow` flag will not work if the user passes a host name that includes `https://`. I went with that decision since it seems like the potential host values found in GH_HOST or in gh CLI configuration do not include the scheme.